adventurine.rb

```diff
@@ -1,26 +1,87 @@
 require 'pp' #so the board prints prettier
 
 board = [
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"]
 ]
 
-user_pos = [0,0]
-a = [-1,0]
-s = [0,1]
-w = [0,-1]
-d = [1,0]
+user_x = 0 
+user_y = 0
+valid_moves = ['w', 'a', 's', 'd', 'x', 'p']
 
-pp board
-print "You are in the northwestern most corner of a dungeon. Use 'a' to move west, 's' to move south, 'w' to move north, and 'd' to move east: "
-user_input = gets.chomp.downcase
+puts "What is your character's name?"
+character = gets.chomp
 
-if user_input == 'a' 
-  user_pos = user_pos.zip(a).map { |x, y| x + y } #if input is a then add user_pos array to a array and store it in user_pos
-elsif user_input == 'b' 
-  user_pos = user_pos.zip(b).map { |x, y| x + y }
-end #same thing if input is b...etc
+board[user_x][user_y] = character.chars[0].upcase
+
+
+board.each do |row|
+  row.each do |column|
+    print "#{column} "
+  end
+  puts ""
+end
+puts "You are in the northwestern most corner of a dungeon."
+puts "Use 'a' to move west, 's' to move south, 'w' to move north, and 'd' to move east"
+puts "Typing 'p' will show your position and 'x' will quit"
+
+loop do
+  puts "WASD?"
+  move = gets.chomp.downcase
+  if valid_moves.include? move
+    if move == 'w' 
+      puts "Walking up..."
+      if user_y - 1 >= 0
```

Comment:
This is to make sure the player doesn't walk off the board. They start in the top left, at 0,0. To go down you have to add one to `user_y`. To go right you add to `user_x`. Up and left are subtracting one. But players can never go below zero and they should never go above the length of the rows array or the columns array of a given row.